src/main/java/Duke.java

```diff
@@ -1,10 +1,247 @@
+import java.time.format.DateTimeParseException;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.stream.Stream;
+
 public class Duke {
+    private ArrayList<Task> db = new ArrayList<>(10);
+    private Scanner sc = new Scanner(System.in);
+
+    /**
+     * Prints welcome message upon starting the bot.
+     */
+    public void greet() {
+        System.out.println("Hello from Duke!\n");
+    }
+
+    /**
+     * Prompts the user for input.
+     * @param sc Scanner object to read user input
+     * @return User input for processing
+     */
+    public String getInput(Scanner sc) {
+        System.out.println("What can I do for you?");
+        return sc.nextLine();
+    }
+
+    /**
+     * Checks the user input and performs the necessary instruction.
+     * @param s User input
+     * @return false if "bye" is input, true for any other instruction
+     */
+    public boolean processInput(String s) {
+        String[] userInput = s.split(" ", 2);
+        boolean goodbye = false;
+
+        try {
+            if (s.equals("bye")) {
+                System.out.println("Bye. Hope to see you again!");
+                goodbye = true;
+                return true;
+            } else if (s.equals("list")) {
+                this.list();
+            } else if (userInput[0].equals("todo")) {
+                this.addTodo(userInput);
+            } else if (userInput[0].equals("deadline")) {
+                this.addDeadline(userInput);
+            } else if (userInput[0].equals("event")) {
+                this.addEvent(userInput);
+            } else if (userInput[0].equals("mark")) {
+                this.mark(userInput);
+            } else if (userInput[0].equals("unmark")) {
+                this.unmark(userInput);
+            } else if (userInput[0].equals("delete")) {
+                this.delete(userInput);
+            } else {
+                throw new DukeException("Unrecognized command.");
+            }
```

Comment:
Perhaps a switch statement would be better here?

Reply:
Yup, just implemented it. Good suggestion!